bubble_sort.rb

```diff
@@ -0,0 +1,34 @@
+# Bubble sort method for sorting numbers
+
+def bubble_sort(arr)
+  arr2 = []
+  arr.length.times do
+    (arr.length - 1).times do
+      arr[0], arr[1] = arr[1], arr[0] if arr[0] <= arr[1]
+      arr.push(arr.shift)
+    end
+    arr2.push(arr.shift)
+  end
+  arr2
+end
```

Comment:
- [ ] bubble_sort method is implemented and working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
- [ ] bubble_sort sorts an array of numbers in increasing order

Reply:
This does not pass yet;
Try this
``def bubble_sort(arr)
   n = arr.length
   loop do
    swapped = false
    (n - 1).times do |i|
      if arr[i] > arr[i + 1]
        arr[i], arr[i + 1] = arr[i + 1], arr[i] # swap the elements
        swapped = true
      end
    end
    break unless swapped
  end
  arr
end``

Reply:
Thanx you for the suggestion @whiz25!